// @vapi-ai/server-sdk/dist/cjs/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.tools = exports.structuredOutputs = exports.squads = exports.sessions = exports.providerResources = exports.phoneNumbers = exports.observabilityScorecard = exports.insight = exports.files = exports.eval = exports.chats = exports.campaigns = exports.calls = exports.assistants = exports.analytics = void 0;
__exportStar(require("./analytics/client/requests/index.js"), exports);
exports.analytics = __importStar(require("./analytics/index.js"));
__exportStar(require("./assistants/client/requests/index.js"), exports);
exports.assistants = __importStar(require("./assistants/index.js"));
__exportStar(require("./assistants/types/index.js"), exports);
__exportStar(require("./calls/client/requests/index.js"), exports);
exports.calls = __importStar(require("./calls/index.js"));
__exportStar(require("./calls/types/index.js"), exports);
__exportStar(require("./campaigns/client/requests/index.js"), exports);
exports.campaigns = __importStar(require("./campaigns/index.js"));
__exportStar(require("./campaigns/types/index.js"), exports);
__exportStar(require("./chats/client/requests/index.js"), exports);
exports.chats = __importStar(require("./chats/index.js"));
__exportStar(require("./chats/types/index.js"), exports);
__exportStar(require("./eval/client/requests/index.js"), exports);
exports.eval = __importStar(require("./eval/index.js"));
__exportStar(require("./eval/types/index.js"), exports);
__exportStar(require("./files/client/requests/index.js"), exports);
exports.files = __importStar(require("./files/index.js"));
__exportStar(require("./insight/client/requests/index.js"), exports);
exports.insight = __importStar(require("./insight/index.js"));
__exportStar(require("./insight/types/index.js"), exports);
__exportStar(require("./observabilityScorecard/client/requests/index.js"), exports);
exports.observabilityScorecard = __importStar(require("./observabilityScorecard/index.js"));
__exportStar(require("./observabilityScorecard/types/index.js"), exports);
__exportStar(require("./phoneNumbers/client/requests/index.js"), exports);
exports.phoneNumbers = __importStar(require("./phoneNumbers/index.js"));
__exportStar(require("./phoneNumbers/types/index.js"), exports);
__exportStar(require("./providerResources/client/requests/index.js"), exports);
exports.providerResources = __importStar(require("./providerResources/index.js"));
__exportStar(require("./providerResources/types/index.js"), exports);
__exportStar(require("./sessions/client/requests/index.js"), exports);
exports.sessions = __importStar(require("./sessions/index.js"));
__exportStar(require("./sessions/types/index.js"), exports);
__exportStar(require("./squads/client/requests/index.js"), exports);
exports.squads = __importStar(require("./squads/index.js"));
__exportStar(require("./structuredOutputs/client/requests/index.js"), exports);
exports.structuredOutputs = __importStar(require("./structuredOutputs/index.js"));
__exportStar(require("./structuredOutputs/types/index.js"), exports);
__exportStar(require("./tools/client/requests/index.js"), exports);
exports.tools = __importStar(require("./tools/index.js"));
__exportStar(require("./tools/types/index.js"), exports);
