// @vapi-ai/server-sdk/dist/cjs/Client.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VapiClient = void 0;
const Client_js_1 = require("./api/resources/analytics/client/Client.js");
const Client_js_2 = require("./api/resources/assistants/client/Client.js");
const Client_js_3 = require("./api/resources/calls/client/Client.js");
const Client_js_4 = require("./api/resources/campaigns/client/Client.js");
const Client_js_5 = require("./api/resources/chats/client/Client.js");
const Client_js_6 = require("./api/resources/eval/client/Client.js");
const Client_js_7 = require("./api/resources/files/client/Client.js");
const Client_js_8 = require("./api/resources/insight/client/Client.js");
const Client_js_9 = require("./api/resources/observabilityScorecard/client/Client.js");
const Client_js_10 = require("./api/resources/phoneNumbers/client/Client.js");
const Client_js_11 = require("./api/resources/providerResources/client/Client.js");
const Client_js_12 = require("./api/resources/sessions/client/Client.js");
const Client_js_13 = require("./api/resources/squads/client/Client.js");
const Client_js_14 = require("./api/resources/structuredOutputs/client/Client.js");
const Client_js_15 = require("./api/resources/tools/client/Client.js");
const BaseClient_js_1 = require("./BaseClient.js");
class VapiClient {
    constructor(options) {
        this._options = (0, BaseClient_js_1.normalizeClientOptions)(options);
    }
    get assistants() {
        var _a;
        return ((_a = this._assistants) !== null && _a !== void 0 ? _a : (this._assistants = new Client_js_2.AssistantsClient(this._options)));
    }
    get squads() {
        var _a;
        return ((_a = this._squads) !== null && _a !== void 0 ? _a : (this._squads = new Client_js_13.SquadsClient(this._options)));
    }
    get calls() {
        var _a;
        return ((_a = this._calls) !== null && _a !== void 0 ? _a : (this._calls = new Client_js_3.CallsClient(this._options)));
    }
    get chats() {
        var _a;
        return ((_a = this._chats) !== null && _a !== void 0 ? _a : (this._chats = new Client_js_5.ChatsClient(this._options)));
    }
    get campaigns() {
        var _a;
        return ((_a = this._campaigns) !== null && _a !== void 0 ? _a : (this._campaigns = new Client_js_4.CampaignsClient(this._options)));
    }
    get sessions() {
        var _a;
        return ((_a = this._sessions) !== null && _a !== void 0 ? _a : (this._sessions = new Client_js_12.SessionsClient(this._options)));
    }
    get phoneNumbers() {
        var _a;
        return ((_a = this._phoneNumbers) !== null && _a !== void 0 ? _a : (this._phoneNumbers = new Client_js_10.PhoneNumbersClient(this._options)));
    }
    get tools() {
        var _a;
        return ((_a = this._tools) !== null && _a !== void 0 ? _a : (this._tools = new Client_js_15.ToolsClient(this._options)));
    }
    get files() {
        var _a;
        return ((_a = this._files) !== null && _a !== void 0 ? _a : (this._files = new Client_js_7.FilesClient(this._options)));
    }
    get structuredOutputs() {
        var _a;
        return ((_a = this._structuredOutputs) !== null && _a !== void 0 ? _a : (this._structuredOutputs = new Client_js_14.StructuredOutputsClient(this._options)));
    }
    get insight() {
        var _a;
        return ((_a = this._insight) !== null && _a !== void 0 ? _a : (this._insight = new Client_js_8.InsightClient(this._options)));
    }
    get eval() {
        var _a;
        return ((_a = this._eval) !== null && _a !== void 0 ? _a : (this._eval = new Client_js_6.EvalClient(this._options)));
    }
    get observabilityScorecard() {
        var _a;
        return ((_a = this._observabilityScorecard) !== null && _a !== void 0 ? _a : (this._observabilityScorecard = new Client_js_9.ObservabilityScorecardClient(this._options)));
    }
    get providerResources() {
        var _a;
        return ((_a = this._providerResources) !== null && _a !== void 0 ? _a : (this._providerResources = new Client_js_11.ProviderResourcesClient(this._options)));
    }
    get analytics() {
        var _a;
        return ((_a = this._analytics) !== null && _a !== void 0 ? _a : (this._analytics = new Client_js_1.AnalyticsClient(this._options)));
    }
}
exports.VapiClient = VapiClient;
