// @vapi-ai/server-sdk/dist/esm/api/types/DeepgramVoiceId.mjs
// This file was auto-generated by Fern from our API Definition.
/** This is the provider-specific ID that will be used. */
export const DeepgramVoiceId = {
    Asteria: "asteria",
    Luna: "luna",
    Stella: "stella",
    Athena: "athena",
    Hera: "hera",
    Orion: "orion",
    Arcas: "arcas",
    Perseus: "perseus",
    Angus: "angus",
    Orpheus: "orpheus",
    Helios: "helios",
    Zeus: "zeus",
    Thalia: "thalia",
    Andromeda: "andromeda",
    Helena: "helena",
    Apollo: "apollo",
    Aries: "aries",
    Amalthea: "amalthea",
    Atlas: "atlas",
    Aurora: "aurora",
    Callista: "callista",
    Cora: "cora",
    Cordelia: "cordelia",
    Delia: "delia",
    Draco: "draco",
    Electra: "electra",
    Harmonia: "harmonia",
    Hermes: "hermes",
    Hyperion: "hyperion",
    Iris: "iris",
    Janus: "janus",
    Juno: "juno",
    Jupiter: "jupiter",
    Mars: "mars",
    Minerva: "minerva",
    Neptune: "neptune",
    Odysseus: "odysseus",
    Ophelia: "ophelia",
    Pandora: "pandora",
    Phoebe: "phoebe",
    Pluto: "pluto",
    Saturn: "saturn",
    Selene: "selene",
    Theia: "theia",
    Vesta: "vesta",
    Celeste: "celeste",
    Estrella: "estrella",
    Nestor: "nestor",
    Sirio: "sirio",
    Carina: "carina",
    Alvaro: "alvaro",
    Diana: "diana",
    Aquila: "aquila",
    Selena: "selena",
    Javier: "javier",
};
