// @vapi-ai/server-sdk/dist/esm/api/types/FallbackPlayHtVoiceLanguage.mjs
// This file was auto-generated by Fern from our API Definition.
/** The language to use for the speech. */
export const FallbackPlayHtVoiceLanguage = {
    Afrikaans: "afrikaans",
    Albanian: "albanian",
    Amharic: "amharic",
    Arabic: "arabic",
    Bengali: "bengali",
    Bulgarian: "bulgarian",
    Catalan: "catalan",
    Croatian: "croatian",
    Czech: "czech",
    Danish: "danish",
    Dutch: "dutch",
    English: "english",
    French: "french",
    Galician: "galician",
    German: "german",
    Greek: "greek",
    Hebrew: "hebrew",
    Hindi: "hindi",
    Hungarian: "hungarian",
    Indonesian: "indonesian",
    Italian: "italian",
    Japanese: "japanese",
    Korean: "korean",
    Malay: "malay",
    Mandarin: "mandarin",
    Polish: "polish",
    Portuguese: "portuguese",
    Russian: "russian",
    Serbian: "serbian",
    Spanish: "spanish",
    Swedish: "swedish",
    Tagalog: "tagalog",
    Thai: "thai",
    Turkish: "turkish",
    Ukrainian: "ukrainian",
    Urdu: "urdu",
    Xhosa: "xhosa",
};
