// @vapi-ai/server-sdk/dist/esm/api/types/FallbackGladiaTranscriberLanguages.mjs
// This file was auto-generated by Fern from our API Definition.
/** Defines the languages to use for the transcription. Required when languageBehaviour is 'manual'. */
export const FallbackGladiaTranscriberLanguages = {
    Af: "af",
    Sq: "sq",
    Am: "am",
    Ar: "ar",
    Hy: "hy",
    As: "as",
    Az: "az",
    Ba: "ba",
    Eu: "eu",
    Be: "be",
    Bn: "bn",
    Bs: "bs",
    Br: "br",
    Bg: "bg",
    Ca: "ca",
    Zh: "zh",
    Hr: "hr",
    Cs: "cs",
    Da: "da",
    Nl: "nl",
    En: "en",
    Et: "et",
    Fo: "fo",
    Fi: "fi",
    Fr: "fr",
    Gl: "gl",
    Ka: "ka",
    De: "de",
    El: "el",
    Gu: "gu",
    Ht: "ht",
    Ha: "ha",
    Haw: "haw",
    He: "he",
    Hi: "hi",
    Hu: "hu",
    Is: "is",
    Id: "id",
    It: "it",
    Ja: "ja",
    Jv: "jv",
    Kn: "kn",
    Kk: "kk",
    Km: "km",
    Ko: "ko",
    Lo: "lo",
    La: "la",
    Lv: "lv",
    Ln: "ln",
    Lt: "lt",
    Lb: "lb",
    Mk: "mk",
    Mg: "mg",
    Ms: "ms",
    Ml: "ml",
    Mt: "mt",
    Mi: "mi",
    Mr: "mr",
    Mn: "mn",
    My: "my",
    Ne: "ne",
    No: "no",
    Nn: "nn",
    Oc: "oc",
    Ps: "ps",
    Fa: "fa",
    Pl: "pl",
    Pt: "pt",
    Pa: "pa",
    Ro: "ro",
    Ru: "ru",
    Sa: "sa",
    Sr: "sr",
    Sn: "sn",
    Sd: "sd",
    Si: "si",
    Sk: "sk",
    Sl: "sl",
    So: "so",
    Es: "es",
    Su: "su",
    Sw: "sw",
    Sv: "sv",
    Tl: "tl",
    Tg: "tg",
    Ta: "ta",
    Tt: "tt",
    Te: "te",
    Th: "th",
    Bo: "bo",
    Tr: "tr",
    Tk: "tk",
    Uk: "uk",
    Ur: "ur",
    Uz: "uz",
    Vi: "vi",
    Cy: "cy",
    Yi: "yi",
    Yo: "yo",
};
