// @vapi-ai/server-sdk/dist/esm/api/types/WorkflowOpenAiModelModel.mjs
// This file was auto-generated by Fern from our API Definition.
/**
 * This is the OpenAI model that will be used.
 *
 * When using Vapi OpenAI or your own Azure Credentials, you have the option to specify the region for the selected model. This shouldn't be specified unless you have a specific reason to do so. Vapi will automatically find the fastest region that make sense.
 * This is helpful when you are required to comply with Data Residency rules. Learn more about Azure regions here https://azure.microsoft.com/en-us/explore/global-infrastructure/data-residency/.
 */
export const WorkflowOpenAiModelModel = {
    Gpt51: "gpt-5.1",
    Gpt51ChatLatest: "gpt-5.1-chat-latest",
    Gpt5: "gpt-5",
    Gpt5Mini: "gpt-5-mini",
    Gpt5Nano: "gpt-5-nano",
    Gpt4120250414: "gpt-4.1-2025-04-14",
    Gpt41Mini20250414: "gpt-4.1-mini-2025-04-14",
    Gpt41Nano20250414: "gpt-4.1-nano-2025-04-14",
    Gpt41: "gpt-4.1",
    Gpt41Mini: "gpt-4.1-mini",
    Gpt41Nano: "gpt-4.1-nano",
    Chatgpt4OLatest: "chatgpt-4o-latest",
    O3: "o3",
    O3Mini: "o3-mini",
    O4Mini: "o4-mini",
    O1Mini: "o1-mini",
    O1Mini20240912: "o1-mini-2024-09-12",
    Gpt4OMini20240718: "gpt-4o-mini-2024-07-18",
    Gpt4OMini: "gpt-4o-mini",
    Gpt4O: "gpt-4o",
    Gpt4O20240513: "gpt-4o-2024-05-13",
    Gpt4O20240806: "gpt-4o-2024-08-06",
    Gpt4O20241120: "gpt-4o-2024-11-20",
    Gpt4Turbo: "gpt-4-turbo",
    Gpt4Turbo20240409: "gpt-4-turbo-2024-04-09",
    Gpt4TurboPreview: "gpt-4-turbo-preview",
    Gpt40125Preview: "gpt-4-0125-preview",
    Gpt41106Preview: "gpt-4-1106-preview",
    Gpt4: "gpt-4",
    Gpt40613: "gpt-4-0613",
    Gpt35Turbo: "gpt-3.5-turbo",
    Gpt35Turbo0125: "gpt-3.5-turbo-0125",
    Gpt35Turbo1106: "gpt-3.5-turbo-1106",
    Gpt35Turbo16K: "gpt-3.5-turbo-16k",
    Gpt35Turbo0613: "gpt-3.5-turbo-0613",
    Gpt4120250414Westus: "gpt-4.1-2025-04-14:westus",
    Gpt4120250414Eastus2: "gpt-4.1-2025-04-14:eastus2",
    Gpt4120250414Eastus: "gpt-4.1-2025-04-14:eastus",
    Gpt4120250414Westus3: "gpt-4.1-2025-04-14:westus3",
    Gpt4120250414Northcentralus: "gpt-4.1-2025-04-14:northcentralus",
    Gpt4120250414Southcentralus: "gpt-4.1-2025-04-14:southcentralus",
    Gpt41Mini20250414Westus: "gpt-4.1-mini-2025-04-14:westus",
    Gpt41Mini20250414Eastus2: "gpt-4.1-mini-2025-04-14:eastus2",
    Gpt41Mini20250414Eastus: "gpt-4.1-mini-2025-04-14:eastus",
    Gpt41Mini20250414Westus3: "gpt-4.1-mini-2025-04-14:westus3",
    Gpt41Mini20250414Northcentralus: "gpt-4.1-mini-2025-04-14:northcentralus",
    Gpt41Mini20250414Southcentralus: "gpt-4.1-mini-2025-04-14:southcentralus",
    Gpt41Nano20250414Westus: "gpt-4.1-nano-2025-04-14:westus",
    Gpt41Nano20250414Eastus2: "gpt-4.1-nano-2025-04-14:eastus2",
    Gpt41Nano20250414Westus3: "gpt-4.1-nano-2025-04-14:westus3",
    Gpt41Nano20250414Northcentralus: "gpt-4.1-nano-2025-04-14:northcentralus",
    Gpt41Nano20250414Southcentralus: "gpt-4.1-nano-2025-04-14:southcentralus",
    Gpt4O20241120Swedencentral: "gpt-4o-2024-11-20:swedencentral",
    Gpt4O20241120Westus: "gpt-4o-2024-11-20:westus",
    Gpt4O20241120Eastus2: "gpt-4o-2024-11-20:eastus2",
    Gpt4O20241120Eastus: "gpt-4o-2024-11-20:eastus",
    Gpt4O20241120Westus3: "gpt-4o-2024-11-20:westus3",
    Gpt4O20241120Southcentralus: "gpt-4o-2024-11-20:southcentralus",
    Gpt4O20240806Westus: "gpt-4o-2024-08-06:westus",
    Gpt4O20240806Westus3: "gpt-4o-2024-08-06:westus3",
    Gpt4O20240806Eastus: "gpt-4o-2024-08-06:eastus",
    Gpt4O20240806Eastus2: "gpt-4o-2024-08-06:eastus2",
    Gpt4O20240806Northcentralus: "gpt-4o-2024-08-06:northcentralus",
    Gpt4O20240806Southcentralus: "gpt-4o-2024-08-06:southcentralus",
    Gpt4OMini20240718Westus: "gpt-4o-mini-2024-07-18:westus",
    Gpt4OMini20240718Westus3: "gpt-4o-mini-2024-07-18:westus3",
    Gpt4OMini20240718Eastus: "gpt-4o-mini-2024-07-18:eastus",
    Gpt4OMini20240718Eastus2: "gpt-4o-mini-2024-07-18:eastus2",
    Gpt4OMini20240718Northcentralus: "gpt-4o-mini-2024-07-18:northcentralus",
    Gpt4OMini20240718Southcentralus: "gpt-4o-mini-2024-07-18:southcentralus",
    Gpt4O20240513Eastus2: "gpt-4o-2024-05-13:eastus2",
    Gpt4O20240513Eastus: "gpt-4o-2024-05-13:eastus",
    Gpt4O20240513Northcentralus: "gpt-4o-2024-05-13:northcentralus",
    Gpt4O20240513Southcentralus: "gpt-4o-2024-05-13:southcentralus",
    Gpt4O20240513Westus3: "gpt-4o-2024-05-13:westus3",
    Gpt4O20240513Westus: "gpt-4o-2024-05-13:westus",
    Gpt4Turbo20240409Eastus2: "gpt-4-turbo-2024-04-09:eastus2",
    Gpt40125PreviewEastus: "gpt-4-0125-preview:eastus",
    Gpt40125PreviewNorthcentralus: "gpt-4-0125-preview:northcentralus",
    Gpt40125PreviewSouthcentralus: "gpt-4-0125-preview:southcentralus",
    Gpt41106PreviewAustralia: "gpt-4-1106-preview:australia",
    Gpt41106PreviewCanadaeast: "gpt-4-1106-preview:canadaeast",
    Gpt41106PreviewFrance: "gpt-4-1106-preview:france",
    Gpt41106PreviewIndia: "gpt-4-1106-preview:india",
    Gpt41106PreviewNorway: "gpt-4-1106-preview:norway",
    Gpt41106PreviewSwedencentral: "gpt-4-1106-preview:swedencentral",
    Gpt41106PreviewUk: "gpt-4-1106-preview:uk",
    Gpt41106PreviewWestus: "gpt-4-1106-preview:westus",
    Gpt41106PreviewWestus3: "gpt-4-1106-preview:westus3",
    Gpt40613Canadaeast: "gpt-4-0613:canadaeast",
    Gpt35Turbo0125Canadaeast: "gpt-3.5-turbo-0125:canadaeast",
    Gpt35Turbo0125Northcentralus: "gpt-3.5-turbo-0125:northcentralus",
    Gpt35Turbo0125Southcentralus: "gpt-3.5-turbo-0125:southcentralus",
    Gpt35Turbo1106Canadaeast: "gpt-3.5-turbo-1106:canadaeast",
    Gpt35Turbo1106Westus: "gpt-3.5-turbo-1106:westus",
};
