// @vapi-ai/server-sdk/dist/cjs/api/types/CreateAssistantDtoServerMessagesItem.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAssistantDtoServerMessagesItem = void 0;
exports.CreateAssistantDtoServerMessagesItem = {
    AssistantStarted: "assistant.started",
    ConversationUpdate: "conversation-update",
    EndOfCallReport: "end-of-call-report",
    FunctionCall: "function-call",
    Hang: "hang",
    LanguageChanged: "language-changed",
    LanguageChangeDetected: "language-change-detected",
    ModelOutput: "model-output",
    PhoneCallControl: "phone-call-control",
    SpeechUpdate: "speech-update",
    StatusUpdate: "status-update",
    Transcript: "transcript",
    TranscriptTranscriptTypeFinal: 'transcript[transcriptType="final"]',
    ToolCalls: "tool-calls",
    TransferDestinationRequest: "transfer-destination-request",
    HandoffDestinationRequest: "handoff-destination-request",
    TransferUpdate: "transfer-update",
    UserInterrupted: "user-interrupted",
    VoiceInput: "voice-input",
    ChatCreated: "chat.created",
    ChatDeleted: "chat.deleted",
    SessionCreated: "session.created",
    SessionUpdated: "session.updated",
    SessionDeleted: "session.deleted",
    CallDeleted: "call.deleted",
    CallDeleteFailed: "call.delete.failed",
};
