// @vapi-ai/server-sdk/dist/cjs/api/types/FallbackAzureSpeechTranscriberLanguage.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.FallbackAzureSpeechTranscriberLanguage = void 0;
/** This is the language that will be set for the transcription. The list of languages Azure supports can be found here: https://learn.microsoft.com/en-us/azure/ai-services/speech-service/language-support?tabs=stt */
exports.FallbackAzureSpeechTranscriberLanguage = {
    AfZa: "af-ZA",
    AmEt: "am-ET",
    ArAe: "ar-AE",
    ArBh: "ar-BH",
    ArDz: "ar-DZ",
    ArEg: "ar-EG",
    ArIl: "ar-IL",
    ArIq: "ar-IQ",
    ArJo: "ar-JO",
    ArKw: "ar-KW",
    ArLb: "ar-LB",
    ArLy: "ar-LY",
    ArMa: "ar-MA",
    ArOm: "ar-OM",
    ArPs: "ar-PS",
    ArQa: "ar-QA",
    ArSa: "ar-SA",
    ArSy: "ar-SY",
    ArTn: "ar-TN",
    ArYe: "ar-YE",
    AzAz: "az-AZ",
    BgBg: "bg-BG",
    BnIn: "bn-IN",
    BsBa: "bs-BA",
    CaEs: "ca-ES",
    CsCz: "cs-CZ",
    CyGb: "cy-GB",
    DaDk: "da-DK",
    DeAt: "de-AT",
    DeCh: "de-CH",
    DeDe: "de-DE",
    ElGr: "el-GR",
    EnAu: "en-AU",
    EnCa: "en-CA",
    EnGb: "en-GB",
    EnGh: "en-GH",
    EnHk: "en-HK",
    EnIe: "en-IE",
    EnIn: "en-IN",
    EnKe: "en-KE",
    EnNg: "en-NG",
    EnNz: "en-NZ",
    EnPh: "en-PH",
    EnSg: "en-SG",
    EnTz: "en-TZ",
    EnUs: "en-US",
    EnZa: "en-ZA",
    EsAr: "es-AR",
    EsBo: "es-BO",
    EsCl: "es-CL",
    EsCo: "es-CO",
    EsCr: "es-CR",
    EsCu: "es-CU",
    EsDo: "es-DO",
    EsEc: "es-EC",
    EsEs: "es-ES",
    EsGq: "es-GQ",
    EsGt: "es-GT",
    EsHn: "es-HN",
    EsMx: "es-MX",
    EsNi: "es-NI",
    EsPa: "es-PA",
    EsPe: "es-PE",
    EsPr: "es-PR",
    EsPy: "es-PY",
    EsSv: "es-SV",
    EsUs: "es-US",
    EsUy: "es-UY",
    EsVe: "es-VE",
    EtEe: "et-EE",
    EuEs: "eu-ES",
    FaIr: "fa-IR",
    FiFi: "fi-FI",
    FilPh: "fil-PH",
    FrBe: "fr-BE",
    FrCa: "fr-CA",
    FrCh: "fr-CH",
    FrFr: "fr-FR",
    GaIe: "ga-IE",
    GlEs: "gl-ES",
    GuIn: "gu-IN",
    HeIl: "he-IL",
    HiIn: "hi-IN",
    HrHr: "hr-HR",
    HuHu: "hu-HU",
    HyAm: "hy-AM",
    IdId: "id-ID",
    IsIs: "is-IS",
    ItCh: "it-CH",
    ItIt: "it-IT",
    JaJp: "ja-JP",
    JvId: "jv-ID",
    KaGe: "ka-GE",
    KkKz: "kk-KZ",
    KmKh: "km-KH",
    KnIn: "kn-IN",
    KoKr: "ko-KR",
    LoLa: "lo-LA",
    LtLt: "lt-LT",
    LvLv: "lv-LV",
    MkMk: "mk-MK",
    MlIn: "ml-IN",
    MnMn: "mn-MN",
    MrIn: "mr-IN",
    MsMy: "ms-MY",
    MtMt: "mt-MT",
    MyMm: "my-MM",
    NbNo: "nb-NO",
    NeNp: "ne-NP",
    NlBe: "nl-BE",
    NlNl: "nl-NL",
    PaIn: "pa-IN",
    PlPl: "pl-PL",
    PsAf: "ps-AF",
    PtBr: "pt-BR",
    PtPt: "pt-PT",
    RoRo: "ro-RO",
    RuRu: "ru-RU",
    SiLk: "si-LK",
    SkSk: "sk-SK",
    SlSi: "sl-SI",
    SoSo: "so-SO",
    SqAl: "sq-AL",
    SrRs: "sr-RS",
    SvSe: "sv-SE",
    SwKe: "sw-KE",
    SwTz: "sw-TZ",
    TaIn: "ta-IN",
    TeIn: "te-IN",
    ThTh: "th-TH",
    TrTr: "tr-TR",
    UkUa: "uk-UA",
    UrIn: "ur-IN",
    UzUz: "uz-UZ",
    ViVn: "vi-VN",
    WuuCn: "wuu-CN",
    YueCn: "yue-CN",
    ZhCn: "zh-CN",
    ZhCnShandong: "zh-CN-shandong",
    ZhCnSichuan: "zh-CN-sichuan",
    ZhHk: "zh-HK",
    ZhTw: "zh-TW",
    ZuZa: "zu-ZA",
};
