// @vapi-ai/server-sdk/dist/cjs/api/types/FallbackOpenAiTranscriberLanguage.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.FallbackOpenAiTranscriberLanguage = void 0;
/** This is the language that will be set for the transcription. */
exports.FallbackOpenAiTranscriberLanguage = {
    Af: "af",
    Ar: "ar",
    Hy: "hy",
    Az: "az",
    Be: "be",
    Bs: "bs",
    Bg: "bg",
    Ca: "ca",
    Zh: "zh",
    Hr: "hr",
    Cs: "cs",
    Da: "da",
    Nl: "nl",
    En: "en",
    Et: "et",
    Fi: "fi",
    Fr: "fr",
    Gl: "gl",
    De: "de",
    El: "el",
    He: "he",
    Hi: "hi",
    Hu: "hu",
    Is: "is",
    Id: "id",
    It: "it",
    Ja: "ja",
    Kn: "kn",
    Kk: "kk",
    Ko: "ko",
    Lv: "lv",
    Lt: "lt",
    Mk: "mk",
    Ms: "ms",
    Mr: "mr",
    Mi: "mi",
    Ne: "ne",
    No: "no",
    Fa: "fa",
    Pl: "pl",
    Pt: "pt",
    Ro: "ro",
    Ru: "ru",
    Sr: "sr",
    Sk: "sk",
    Sl: "sl",
    Es: "es",
    Sw: "sw",
    Sv: "sv",
    Tl: "tl",
    Ta: "ta",
    Th: "th",
    Tr: "tr",
    Uk: "uk",
    Ur: "ur",
    Vi: "vi",
    Cy: "cy",
};
