// @vapi-ai/server-sdk/dist/cjs/api/types/GoogleTranscriberLanguage.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleTranscriberLanguage = void 0;
/** This is the language that will be set for the transcription. */
exports.GoogleTranscriberLanguage = {
    Multilingual: "Multilingual",
    Arabic: "Arabic",
    Bengali: "Bengali",
    Bulgarian: "Bulgarian",
    Chinese: "Chinese",
    Croatian: "Croatian",
    Czech: "Czech",
    Danish: "Danish",
    Dutch: "Dutch",
    English: "English",
    Estonian: "Estonian",
    Finnish: "Finnish",
    French: "French",
    German: "German",
    Greek: "Greek",
    Hebrew: "Hebrew",
    Hindi: "Hindi",
    Hungarian: "Hungarian",
    Indonesian: "Indonesian",
    Italian: "Italian",
    Japanese: "Japanese",
    Korean: "Korean",
    Latvian: "Latvian",
    Lithuanian: "Lithuanian",
    Norwegian: "Norwegian",
    Polish: "Polish",
    Portuguese: "Portuguese",
    Romanian: "Romanian",
    Russian: "Russian",
    Serbian: "Serbian",
    Slovak: "Slovak",
    Slovenian: "Slovenian",
    Spanish: "Spanish",
    Swahili: "Swahili",
    Swedish: "Swedish",
    Thai: "Thai",
    Turkish: "Turkish",
    Ukrainian: "Ukrainian",
    Vietnamese: "Vietnamese",
};
