// @vapi-ai/server-sdk/dist/cjs/api/types/InworldVoiceVoiceId.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.InworldVoiceVoiceId = void 0;
/**
 * Available voices by language:
 * • en: Alex, Ashley, Craig, Deborah, Dennis, Edward, Elizabeth, Hades, Julia, Pixie, Mark, Olivia, Priya, Ronald, Sarah, Shaun, Theodore, Timothy, Wendy, Dominus
 * • zh: Yichen, Xiaoyin, Xinyi, Jing
 * • nl: Erik, Katrien, Lennart, Lore
 * • fr: Alain, Hélène, Mathieu, Étienne
 * • de: Johanna, Josef
 * • it: Gianni, Orietta
 * • ja: Asuka, Satoshi
 * • ko: Hyunwoo, Minji, Seojun, Yoona
 * • pl: Szymon, Wojciech
 * • pt: Heitor, Maitê
 * • es: Diego, Lupita, Miguel, Rafael
 */
exports.InworldVoiceVoiceId = {
    Alex: "Alex",
    Ashley: "Ashley",
    Craig: "Craig",
    Deborah: "Deborah",
    Dennis: "Dennis",
    Edward: "Edward",
    Elizabeth: "Elizabeth",
    Hades: "Hades",
    Julia: "Julia",
    Pixie: "Pixie",
    Mark: "Mark",
    Olivia: "Olivia",
    Priya: "Priya",
    Ronald: "Ronald",
    Sarah: "Sarah",
    Shaun: "Shaun",
    Theodore: "Theodore",
    Timothy: "Timothy",
    Wendy: "Wendy",
    Dominus: "Dominus",
    Yichen: "Yichen",
    Xiaoyin: "Xiaoyin",
    Xinyi: "Xinyi",
    Jing: "Jing",
    Erik: "Erik",
    Katrien: "Katrien",
    Lennart: "Lennart",
    Lore: "Lore",
    Alain: "Alain",
    Helene: "H\u00E9l\u00E8ne",
    Mathieu: "Mathieu",
    Etienne: "\u00C9tienne",
    Johanna: "Johanna",
    Josef: "Josef",
    Gianni: "Gianni",
    Orietta: "Orietta",
    Asuka: "Asuka",
    Satoshi: "Satoshi",
    Hyunwoo: "Hyunwoo",
    Minji: "Minji",
    Seojun: "Seojun",
    Yoona: "Yoona",
    Szymon: "Szymon",
    Wojciech: "Wojciech",
    Heitor: "Heitor",
    Maite: "Mait\u00EA",
    Diego: "Diego",
    Lupita: "Lupita",
    Miguel: "Miguel",
    Rafael: "Rafael",
};
