// @vapi-ai/server-sdk/dist/cjs/api/types/MinimaxVoiceLanguageBoost.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MinimaxVoiceLanguageBoost = void 0;
/** Language hint for MiniMax T2A. Example: yue (Cantonese), zh (Chinese), en (English). */
exports.MinimaxVoiceLanguageBoost = {
    Chinese: "Chinese",
    ChineseYue: "Chinese,Yue",
    English: "English",
    Arabic: "Arabic",
    Russian: "Russian",
    Spanish: "Spanish",
    French: "French",
    Portuguese: "Portuguese",
    German: "German",
    Turkish: "Turkish",
    Dutch: "Dutch",
    Ukrainian: "Ukrainian",
    Vietnamese: "Vietnamese",
    Indonesian: "Indonesian",
    Japanese: "Japanese",
    Italian: "Italian",
    Korean: "Korean",
    Thai: "Thai",
    Polish: "Polish",
    Romanian: "Romanian",
    Greek: "Greek",
    Czech: "Czech",
    Finnish: "Finnish",
    Hindi: "Hindi",
    Bulgarian: "Bulgarian",
    Danish: "Danish",
    Hebrew: "Hebrew",
    Malay: "Malay",
    Persian: "Persian",
    Slovak: "Slovak",
    Swedish: "Swedish",
    Croatian: "Croatian",
    Filipino: "Filipino",
    Hungarian: "Hungarian",
    Norwegian: "Norwegian",
    Slovenian: "Slovenian",
    Catalan: "Catalan",
    Nynorsk: "Nynorsk",
    Tamil: "Tamil",
    Afrikaans: "Afrikaans",
    Auto: "auto",
};
